packages/statemanager/src/ethersStateManager.ts

```diff
@@ -167,7 +169,11 @@ export class EthersStateManager extends BaseStateManager implements StateManager
       this.storageCache.set(address.toString(), new Map<string, Buffer>())
       accountStorage = this.storageCache.get(address.toString())
     }
-    accountStorage?.set(key.toString('hex'), value)
+    if (value.length === 0 || value.equals(Buffer.alloc(value.length).fill(0))) {
```

Comment:
More dyslexia -- "delete value" does not mean delete key, just to replace value with empty buffer.
This still needed a tweak, because it did not account for buffers full of zeros triggering a deletion.

I believe the reason there is to prevent users from stockpiling storage space with buffers full of nothing

Reply:
> because it did not account for buffers full of zeros triggering a deletion.
>I believe the reason there is to prevent users from stockpiling storage space with buffers full of nothing

Good catch.  I think you're right.  This is where gas tokens come from most likely.